lib/MetaCPAN/Script/Release.pm

```diff
@@ -490,6 +498,73 @@ sub import_tarball {
     }
 }
 
+sub find_main_module {
+    my $self = shift;
+    my ( $mod, $distribution ) = @_;
+
+    my @modules = @{$mod};
+
+    my $min_level          = 100;
+    my @distribution_parts = split /-/, $distribution;
+    my $distribution_name  = join '', @distribution_parts;
```

Comment:
If you want to compare the distribution name to the module name, you could probably get a comparison variable without the split + join: 

```
my $dist2module = $distribution;
$dist2module =~ s{-}{::}g;
```

Now `HTML-Restrict` becomes `HTML::Restrict`.  If that matches a module, you've got your main_module.